File: C++/Medium/WrongAnswer/longest-happy-string_5.cpp
class Solution {
public:
    string longestDiverseString(int a, int b, int c) {
        
        string res = "";
        vector<pair<int, char>> v;
        v.push_back(make_pair(a, 'a'));
        v.push_back(make_pair(b, 'b'));
        v.push_back(make_pair(c, 'c'));
        while (true) {
            sort(v.begin(), v.end());
            if (v[2].first == 0) break;
            if (v[2].first >= 2 && v[1].first == 0) {
                res += string(2, v[2].second);
                v[2].first -= 2;
            } else if (v[2].first >= 1 && v[1].first >= 1) {
                res += v[2].second;
                res += v[1].second;
                v[2].first -= 1;
                v[1].first -= 1;
            } else if (v[2].first >= 1 && v[1].first == 0) {
                res += v[2].second;
                v[2].first -= 1;
            }
        }
        return res;
    }
};

File: Rust/Medium/WrongAnswer/longest-happy-string_5.rs
impl Solution {
    pub fn longest_diverse_string(a: i32, b: i32, c: i32) -> String {
        
        let mut res = String::new();
        let mut last = 0;
        let mut a = a;
        let mut b = b;
        let mut c = c;
        while a > 0 || b > 0 || c > 0 {
            if a >= b && a >= c && last != 1 {
                res.push('a');
                last = 1;
                a -= 1;
            } else if b >= a && b >= c && last != 2 {
                res.push('b');
                last = 2;
                b -= 1;
            } else if c >= a && c >= b && last != 3 {
                res.push('c');
                last = 3;
                c -= 1;
            } else if a > 0 && last != 1 {
                res.push('a');
                last = 1;
                a -= 1;
            } else if b > 0 && last != 2 {
                res.push('b');
                last = 2;
                b -= 1;
            } else if c > 0 && last != 3 {
                res.push('c');
                last = 3;
                c -= 1;
            } else {
                break;
            }
            if res.len() > 1 && res.chars().nth(res.len()-2).unwrap() == res.chars().nth(res.len()-1).unwrap() {
                res.pop();
                last = 0;
            }
        }
        res
    }
}